### DISCUSSION TITLE
changing branch

### DISCUSSION THREAD

**Discussion by riloraspopo** [end-4/dots-hyprland#108] [General]
i think this is not the bug or error so i dont insert the issue..
how can i change to another branch, right now i'm using m3ww how can i move to illogical_impulse..? or i must clean install the OS..?
i think it's working if i delete folder .config hyprland then install new branch, but there is still apps installed that not used anymore, like eww maybe or something else...

**Comment by end-4** [comment]
just proceed to install the new branch normally. overwrite old files if you no longer need them. else, backup
there's no need to do a full reinstall
if you mind having a few extra packages, look at the instructions of the old branch again for the dependencies and uninstall them

**Comment by clsty** [comment]
There's an install script which will install all apps it needs for illogical-impulse.

You can delete any apps and re-run the script since it's idempotent, and it will always try to ensure those needed apps are ready.

However, we don't have and won't have an uninstall script to handle app removal automatically. Refer to the doc [here](https://end-4.github.io/dots-hyprland-wiki/en/i-i/01setup/#uninstalling):

> The point is that this repo is not a typical “software”, but a set of config, Therefore it’s very hard to make a proper uninstallation script, which should revert all changes made by the installation script.
>
> For example, the installation script will install yay (AUR-helper) for you if you don’t have one. However, what should the uninstallation script do to revert this changes?
>
> - Remove yay or yay-bin? Not proper, because you may already have one of them installed by yourself (NOT by the installation script).
> - Even if the installation script had logged the package list installed by the script, it’s still not proper to remove yay or yay-bin when the log showed yay or yay-bin is installed by the script, because the script can’t log its future, and you may have reinstalled yay or yay-bin another day after the installation script finished, thus they are actually not installed by installation script anymore.
>
> In conclusion, it’s nearly not possible to write a proper uninstallation script. You’d better make reverted changes manually as you need.

Closing.
